### ISSUE TITLE
State of the project

### ISSUE THREAD

**Issue by weisJ** [blackears/svgSalamander#104]
Just wanted to ask what the state of the project is? Do you plan to work on new features or is it in some "maintenance only" state?

**Comment by blackears** [comment]
It's been in maintenance only for some time now.  There are some issues that can't be addressed without a big overhaul of the code.  It would take a few months of work to do that and I can't really justify that especially since I don't really use Java anymore.  (I used to be big fan, but Sun Microsystems and then Oracle have shown no interest in making Java viable on the client side).  Oracle also seems to be pushing Java in a closed source direction these days.  I started this project nearly 20 years ago to be a UI for a game, but Java is a very unrealistic language for game dev in 2023.  

There also seems to be another open source Java svg renderer.  I've not used it, but I presume that is what most Java users are using these days.

Also, sorry for not getting back to you about your offer to help with getting the build process working better.  Some things came up and then I forgot about it.

**Comment by weisJ** [comment]
Thank you for the insight. I totally understand your situation.

> There are some issues that can't be addressed without a big overhaul of the code.

Yeah while working on text layout and masks I also noticed that implementations of some things wouldn't be possible in the current state.

> Also, sorry for not getting back to you about your offer to help with getting the build process working better. Some things came up and then I forgot about it.

Don't worry it was just a suggestion, nothing that was particularily very important to me.

> There also seems to be another open source Java svg renderer. I've not used it, but I presume that is what most Java users are using these days.

Yeah the biggest alternative right now (which I am aware of) is Batik, which also comes with its own shortcomings e.g. it is a rather hefty dependency for "just rendering some svgs".

I was mainly asking because I have been working on my own svg implementation for fun some time now (https://github.com/weisJ/jsvg). It isn't really feature complete with svgSalamander - in the sense that in some areas it supports more of the spec, but is also missing some things. In particular picking elements is currently not possible and animations aren't implemented at all. 

If you say that you don't plan on picking up this project any time soon, I'll see to prioritize implementations of features people coming from svgSalamander want to use, so jsvg can become an alternative in the future.

Thanks btw for all your work on this library. Without it I'd never have thought that implementing something like it to be possible at all :)

**Comment by blackears** [comment]
Yeah, I had hopes that I could turn this into a business of some sort or at least get consulting leads from it, but nothing ever came of it.  One of the big things I would want to do in a redesign is to add variants to the code base.  I didn't realize when I first designed SVGSalamander that in SVG fields can take on multiple data types and this led to a lot of kludges.  (I didn't even know what variants were back then).
